kore/src/main/scala/org/kframework/definition/outer-ext.scala

```diff
@@ -61,7 +63,32 @@ object FlatModule {
   }
 }
 
-case class FlatModule(name: String, imports: Set[String], localSentences: Set[Sentence], @(Nonnull@param) val att: Att = Att.empty)
+case class Import(name: String, att: Att = Att.empty) extends HasLocation {
+  override def location(): Optional[Location] = att.getOptional(classOf[Location])
+  override def source(): Optional[Source] = att.getOptional(classOf[Source])
+}
+
+object Import {
+  val syntaxString = "$SYNTAX"
+
+  def isSyntax(name: String): Boolean = name.endsWith(syntaxString)
+
+  def asSyntax(_import: Import): Import =
+    if (isSyntax(_import.name))
+      _import
+    else
+      Import(_import.name ++ syntaxString, _import.att)
+
+  def noSyntax(name: String): String =
+    if (isSyntax(name))
+      name.dropRight(syntaxString.length)
+    else
+      name
+}
+
+
+
+case class FlatModule(name: String, imports: Set[Import], localSentences: Set[Sentence], @(Nonnull@param) val att: Att = Att.empty)
```

Comment:
Because the JSON encoding is directly an encoding of the `FlatModule`, I think it should be a faithful encoding of the actual `Import`, not just the name.